src/formatter/ExpressionFormatter.ts

```diff
@@ -45,12 +45,21 @@ interface ExpressionFormatterParams {
 export interface DialectFormatOptions {
   // List of operators that should always be formatted without surrounding spaces
   alwaysDenseOperators?: string[];
+  // List of clauses that should be formatted on a single line
+  onelineClauses: string[];
+}
+
+// Contains the same data as DialectFormatOptions,
+// but optimized for faster and more conventient lookup.
+export interface ProcessedDialectFormatOptions {
```

Comment:
can we not just use this as the default interface for DialectFormatOptions and add a wrapper function to transform the string[] into Record ?

Reply:
It's not only the `onlineClauses` field that differs. It's also the `alwaysDenseOperators` field which is originally optional, but then processed to be empty array if missing.